src/status_im/contexts/wallet/common/utils/wallet_connect.cljs

```diff
@@ -0,0 +1,21 @@
+(ns status-im.contexts.wallet.common.utils.wallet-connect
+  (:require ["@walletconnect/core" :refer [Core]]
+            ["@walletconnect/web3wallet" :refer [Web3Wallet]]
+            [promesa.core :as promesa]
+            [status-im.config :as config]))
+
+(defn- wallet-connect-core
+  []
+  (Core. (clj->js {:project-id config/wallet-connect-project-id})))
+
+(defn init
+  []
+  (let [core     (wallet-connect-core)
+        metadata (clj->js config/default-wallet-connect-metadata)]
+    (promesa/create
+     (fn [p-resolve p-reject]
+       (-> (Web3Wallet.init
+            (clj->js {:core     core
```

Comment:
@briansztamfater, a few suggestions for this namespace:

1. The code `(clj->js {:project-id config/wallet-connect-project-id})` can be changed to `#js {:project-id config/WALLET_CONNECT_PROJECT_ID}`. I tend to prefer the `#js` literal for small, non-nested maps because it's significantly faster than `clj->js`. The `init` function will be called once, so it won't be a problem, but sharing as a reminder that we should be mindful of `clj->js`. 
2. I would make `default-wallet-connect-metadata` a private var in this namespace and remove it from `status-im.config` because the map doesn't depend on any value coming from an external env.

Reply:
Updated 👍